### PR TITLE
Avoid querying inventory default stock view in storefront operations

### PR DESCRIPTION
**Pull request by ishakhsuvarov**
### Description (*)
This Pull Request is supposed to reduce the usage of `inventory_stock_1` database view in frontend operations. 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by ishakhsuvarov** [comment]
Looking for a maintainer to review.

File: app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php [hunk 1]
```diff
@@ -28,35 +30,62 @@ class GetStockItemData implements GetStockItemDataInterface
...
-            ->from(
+        $select = $connection->select();
+
+        if ($this->defaultStockProvider->getId() === $stockId) {
+            $productId = current($this->getProductIdsBySkus->execute([$sku]));
+            $stockItemTableName = $this->resource->getTableName('cataloginventory_stock_status');
+            $select->from(
+                $stockItemTableName,
+                [
+                    GetStockItemDataInterface::QUANTITY => 'qty',
```
Comment (swnsma) [inline app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php:75]: Constants may be used 
\Magento\CatalogInventory\Api\Data\StockStatusInterface::QTY
\Magento\CatalogInventory\Api\Data\StockStatusInterface::STOCK_STATUS

File: app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php [hunk 2]
```diff
@@ -28,35 +30,62 @@ class GetStockItemData implements GetStockItemDataInterface
...
+                    GetStockItemDataInterface::IS_SALABLE => 'stock_status',
+                ]
+            )->where('product_id' . ' = ?', $productId);
```
Comment (swnsma) [inline app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php]: I do not see sense to join both stings when you can directly use joined string.

**Review by swnsma — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php (hunk 1 above)
Comment (ishakhsuvarov) [reply app/code/Magento/InventoryIndexer/Model/ResourceModel/GetStockItemData.php:75]: Solution requires adding a dependency on the `CatalogInventory` this should be done in a centralized way, so filed a separate ticket for it.
https://github.com/magento/inventory/issues/2387

**Pull request merged**